srkServer/srkproto/srkproto.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+
+package srkproto;
+
+service FunctionService {
+    
+
+    // ByteTransfer must represent a tar file with a single top-level directory.
+    // The context should include metadata with the following fields:
+    // Required: name - the name of the function being packaged
+    // Optional: includes - comma-separated list of srk packages to include
+    //                      with the function
+    rpc Package(stream ByteTransfer) returns (PackageRet) {}
+
+    // Install performs the srk.FaaS.Install operation (see package srk for details)
+    rpc Install(InstallArg) returns (InstallRet) {}    
+
+    // Invoke the function once and return the response
+    rpc Invoke(InvokeArg) returns (InvokeRet) {}
+
+    rpc Remove(RemoveArg) returns (RemoveRet) {}
+}
+
+message InstallArg {
+    string name = 1;
+}
+
+message InvokeArg {
+    string name = 1;
+
+    // Function argument (json)
+    string farg = 2;
```

Comment:
Should this be bytes? That would be more consistent with the response type.

Reply:
I'm not against this in principle since it allows maximum flexibility (even though all current FaaS implementations I know of require string args). To change this we would also need to change the srk interface (FunctionService) since this API is designed to mirror that as closely as possible.

Reply:
I think we should go with bytes. I find that sometimes the .proto file is a clearer way of spelling out the API than the code. I agree that we should change the FunctionService interface but we might want to do that in another PR. I'm a little bit concerned about merge conflicts at the moment.

Reply:
done. I didn't change the srk package, just the protobuf definition.